### ISSUE TITLE
`replace-newlines` Not Replacing Windows New Lines on Unix

### ISSUE THREAD

**Issue by Silvenga** [NLog/NLog#3937]
**NLog version**: `1.6.8`

**Platform**: .Net 3.5 / .Net 4.5 / .NET Core 2 / .NET Core 3

**Current NLog config**

```csharp
var fileTarget = new FileTarget
{
    FileName = targetPath,
    Encoding = Encoding.UTF8,
    ArchiveAboveSize = 104_857_600, // 100mb
    ArchiveFileName = archivePath,
    KeepFileOpen = true,
    ConcurrentWrites = false,
    OpenFileCacheTimeout = 30,
    EnableFileDelete = true,
    MaxArchiveFiles = 10,
    WriteBom = false,
    Layout = "${longdate} \t"
             + "${level:uppercase=true} \t"
             + "${processid}:${threadid} \t"
             + "${logger} \t"
             + @"${replace-newlines:replacement=\\r\\n:${message}}${onexception:inner= Exception\:${exception:format=ToString}}"
};
```

- What is the current result?

Given the following message logged under Linux (using Windows line endings, per spec):

```
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8" />
    <me...
```

The following is logged:
```
<!DOCTYPE html>
\r\n<html lang="en">
\r\n<head>
\r\n    <meta charset="utf-8" />
\r\n    <me...
```

![image](https://user-images.githubusercontent.com/2753330/80734692-27493e00-8ad5-11ea-8c49-e2c91dca7173.png)

It appears the `\r` character is never replaced.

- What is the expected result?

The replace-newlines wrapper should replace all line endings.

- Are there any workarounds? Regex, the solution to all.
- Is there a version in which it did work? Unknown
- Can you help us by writing an unit test? Yep, could also make a pr, but since this behavior depends on Environment.NewLine, I'm unsure how we want to make that test.

**Comment by Silvenga** [comment]
I can make a pr, I just wanted to check on how you want the unit tests done. I can abstract out `Environment.NewLine`? If these tests run on both Linux and Windows, I could make the new test only execute on Linux?

**Comment by 304NotModified** [comment]
PR would be great. We run on Windows and Linux. Another (maybe better) option would be to mock the newlines values so we could run them everywhere

**Comment by 304NotModified** [comment]
There is already support for Unix/Mac is, but I guess it's buggy then

https://github.com/NLog/NLog/blob/dev/src/NLog/LayoutRenderers/Wrappers/String%20transformations/ReplaceNewLinesLayoutRendererWrapper.cs#L78

**Comment by snakefoot** [comment]
The support is for handling unix-newlines on the windows-platform.

Now it is the other way around. Windows-newlines on the linux-platform (currently unsupported).

**Comment by snakefoot** [comment]
Closing as resolved by  #3949
